pkg/workflow/workflow.go

```diff
@@ -460,12 +471,16 @@ func (e *engine) runAsDAG(taskRunners []wfTypes.TaskRunner) error {
 
 }
 
+func (e *engine) runAsStepByStep(taskRunners []wfTypes.TaskRunner) error {
```

Comment:
Why wrap this function second time? We can use `e.steps(e.todoByIndex(taskRunners))` instead.

Reply:
Because I need to analyse step-group status by sub-steps and update it to the engine.